testsuite/features/step_definitions/vm_steps.rb

```diff
@@ -0,0 +1,410 @@
+# Copyright (c) 2023 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+### This file contains all steps concerning virtual machine management
+
+When(/^I create a (leap|sles|rhlike|deblike) virtual machine named "([^"]*)" (with|without) cloudinit on "([^"]*)"$/) do |os_type, vm_name, cloudinit, host|
+  node = get_target(host)
+  disk_path = "/tmp/#{vm_name}_disk.qcow2"
+
+  # The cloud init images are generated via sumaform and can be modified there
+  # The names of the qcow2 images are also defined there
+  cloudinit_path = "/tmp/cloudinit-disk-#{os_type}.iso"
+  memory = '512'
+
+  case os_type
+  when 'leap'
+    name = 'leap-disk-image-template.qcow2'
+    net =
+      if cloudinit == 'with'
+        'salt-leap'
+      else
+        'test-net0'
+      end
+    os = 'opensuse15.4'
+    memory = '1024'
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
+  when 'sles'
+    name = 'sles-disk-image-template.qcow2'
+    net = 'salt-sles'
+    os = 'sle15sp4'
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
+  when 'rhlike'
+    name = 'rhlike-disk-image-template.qcow2'
+    net = 'salt-rhlike'
+    os = 'rocky8'
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
+  when 'deblike'
+    name = 'deblike-disk-image-template.qcow2'
+    net = 'salt-deblike'
+    os = 'ubuntu2204'
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
+  else
+    name = 'empty'
+  end
+  raise "/var/testsuite-data/#{name} not found" unless file_exists?(node, "/var/testsuite-data/#{name}")
+
+  node.run("cp /var/testsuite-data/#{name} #{disk_path}")
+
+  # Actually define the VM, but don't start it
+  raise 'not found: virt-install' unless file_exists?(node, '/usr/bin/virt-install')
+
+  # with cloud init image
+  if cloudinit == 'with'
+    raise "#{cloudinit_image} not found" unless file_exists?(node, cloudinit_image)
+
+    node.run("cp #{cloudinit_image} #{cloudinit_path}")
+    node.run(
+      "virt-install --name #{vm_name} --memory #{memory} --vcpus 1 --disk path=#{disk_path},bus=virtio " \
+      "--network network=#{net} --graphics vnc,listen=0.0.0.0 " \
+      "--serial file,path=/tmp/#{vm_name}.console.log " \
+      "--import --hvm --noautoconsole --noreboot --osinfo #{os} " \
```

Comment:
I would extract this part to a string variable. To don't have code duplication.
But not a big deal.